x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/percentiles_agg/config.ts

```diff
@@ -71,13 +67,13 @@ export function getPercentilesAggConfig(
 
       return {
         field: this.field as string,
-        percents: parsePercentsInput(this.aggConfig.percents),
+        percents: this.aggConfig.percents ?? [],
       };
     },
     isValid() {
-      return (
-        typeof this.aggConfig.percents === 'string' && isValidPercentsInput(this.aggConfig.percents)
-      );
+      const validationResult = validatePercentsInput(this.aggConfig);
+      this.errorMessageType = validationResult.errorType;
```

Comment:
Added this side effect to prevent `validatePercentsInput` from being called twice, just to extract the error message in a separate function.